projects/js-packages/shared-extension-utils/src/get-block-icon-from-metadata.js

```diff
@@ -42,15 +42,19 @@ export function getBlockIconComponent( metadata ) {
 /**
  * A block icon needs to be redefined on the front end as a React component, since a string - even
  * SVG markup - is interpreted as a dashicon. This function returns the object that must be passed
- * to the `icon` attribute when registering the block in the front end. It also sets the color
- * of the icon.
+ * to the `icon` attribute when registering the block in the front end. For native, this isn't necessary
+ * as there is logic for parsing the string in Gutenberg's `/components/block-icon/index.native.js`
+ * file.
+ *
+ * This function also sets the color of the icon.
  *
  * @param {object} metadata - Block.json content
  * @returns {object} Icon property for client registration
  */
 export function getBlockIconProp( metadata ) {
+	const isNative = Platform.isNative;
 	return {
-		src: getBlockIconComponent( metadata ),
+		src: isNative ? metadata.icon : getBlockIconComponent( metadata ),
```

Comment:
Alternatively, we could create the native version of this file:

```
import getIconColor from './get-icon-color';

/**
 * Provides needed icon props to render the block icon, including the color.
 *
 * @param {object} metadata - Block.json content
 * @returns {object} Icon property for client registration
 */
export function getBlockIconProp( metadata ) {
	return {
		src: metadata.icon,
		foreground: getIconColor(),
	};
}
```

This option might give us more flexibility in the future, plus reduce friction in case there are changes in the web version that affect the native one. WDYT?

Reply:
I don't feel overly strongly, but my thought process behind this approach was that it may help avoid the impact of changes on native being overlooked. I'm a bit worried that the native files were overlooked when the changes to block registration happened, so thought that trying to keep native logic in the main files when possible could help mitigate that in the future.

I'm open to separating this logic out, however.

Reply:
Yep, good point. Definitely, each approach has its benefits and drawbacks. In terms of making more prominent the potential impact on the native version when introducing changes, I'd rather flag them via unit/E2E tests. Since we don't have them set in the Jetpack repository, sharing code in files might be our best option. That said, I agree on continuing with your approach 👍 .

Reply:
For posterity, sharing that I went ahead to create a separate `get-block-icon-from-metadata.native.js` file, as we started to make more native-specific changes that I thought could have caused confusion if kept in the same file.